### ISSUE TITLE
Position on the OpenD compiler

### ISSUE THREAD

**Issue by SingingBush** [intellij-dlanguage/intellij-dlanguage#932]
it's early days for the [OpenD](https://github.com/opendlang/opend/) fork but I think it's worth stating my current position on it for the purposes of this project. 

I don't have personal need to use the fork so I'm not likely to try using it any time soon. I'm not against minor changes being made for the binary to be recognised as a D compiler (as it may not work out of the box), but I don't currently have plans to look into it. 

If it turns out that D and OpenD start to diverge significantly then I'll likely be apposed to making any changes to support it. Keeping up to date with changes to the D language, phobos, 3 existing compilers, Kotlin, the Intellij Platform, etc is already time consuming enough. 

I wish the opendlang team luck with their project. Perhaps if it gains significant adoption my stance may change in the future but for now I'm going to remain cautious about it.

**Comment by iz0eyj** [comment]
I'm only getting closer to the D language these days, everything is new to me... both the language itself and its world.
I confess that this fork left me a little perplexed, both in its aims and reasons.
Perhaps it depends on the fact that the new initiative is very young, but its goal is not at all clear to me.
Many decades of software development lead me to see standardization as a value and, conversely, fragmentation as a disincentive to further adopt a paradigm.
The D language seems excellent to me for people like me who come from C and C#... a modernity similar to C# but with an efficiency comparable to C, great!
However, despite this advantage, its adoption is modest... the interest that Rust has (rightly) aroused has not been created around D.
So nothing against forks, if they have a cause and purpose that objectively justify them, EVERYTHING against forks if they are based solely on a desire for protagonism.
I'm too new to this world to know which of the two cases OpenD falls into.
If the reasons are real and by deviating from the main trend it will be possible to achieve a concrete improvement of the project, then it is correct to follow the new evolution, otherwise doing so would mean leading to an unjustified fragmentation.
I would like to point out one thing: good projects, especially if they are large in size, arise from an analysis carried out a priori from which both the starting point and the goal must be clear.
Now, we know the starting point, it is the D language, but what about the arrival point?
From the fork of a language I would first expect the definition of its variant, because otherwise the fork becomes "we'll make it do something different, but we don't know what yet".

My two cents.

**Comment by schveiguy** [comment]
If this is your post @iz0eyj https://forum.dlang.org/post/vyiwbbsrfnesskoxlwam@forum.dlang.org

> And that's not all, because the maintainer of the Jetbrains plugin declares on GitHub that he wants to follow OpenD.

I think I can say you misread something. @SingingBush says above that he does *not* have plans to support OpenD but is willing to accept changes as long as they are not disruptive (correct me if I'm wrong)

I think you can safely ignore openD for now.

**Comment by SingingBush** [comment]
Yes, I don't know much about opend but if for instance the binary names differ then it would require a change. If someone that wants to use it raised a PR with minimal changes that will get it detected as a valid compiler then that's the kind of thing that could be accepted. It would mean fairly small changes and shouldn't impact the rest of the code. I don't think we should be merging parser changes for supporting any language features that are not in D itself.

**Comment by iz0eyj** [comment]
> If this is your post @iz0eyj https://forum.dlang.org/post/vyiwbbsrfnesskoxlwam@forum.dlang.org
> 
> > And that's not all, because the maintainer of the Jetbrains plugin declares on GitHub that he wants to follow OpenD.
> 
> I think I can say you misread something. @SingingBush says above that he does _not_ have plans to support OpenD but is willing to accept changes as long as they are not disruptive (correct me if I'm wrong)
> 
> I think you can safely ignore openD for now.

**" Perhaps if it gains significant adoption my stance may change in the future but for now I'm going to remain cautious about it."**

This sentence doesn't seem like a no to me, but a "maybe" conditioned on the number of adopters.
A firm no, which would be a heavy rejection since there are only a few plugins for using D in professional development environments (this one and the one for Visual Studio), would discourage both potential users and the authors of the fork themselves.
On the contrary, a "maybe" motivates the search for a sufficient number of developers.
But I'm not even the spare tire on the wagon, so my opinion matters little.

**Comment by SingingBush** [comment]
I've done some playing around with OpenD. With some tweaks we can probably get it to be a compiler option when we add support for gdc and ldc. However, dub doesn't support OpenD. 

I opened https://github.com/dlang/dub/issues/3092 to ask about adding opend support. I have it working locally but it's a bit rough around the edges. I've not tested compiler flags at all. Did get a build of dub that will compile a project with `--compiler=opend` though

**Comment by SingingBush** [comment]
the next release will allow opend to be selected as a D compiler. See #1242
